src/i18n.ts

```diff
@@ -1,40 +1,89 @@
+import ro from "./locales/ro/translation.json";
+import ru from "./locales/ru/translation.json";
+import sr from "./locales/sr/translation.json";
+import ta from "./locales/ta/translation.json";
+import tl from "./locales/tl/translation.json";
+import tr from "./locales/tr/translation.json";
+import zhCN from "./locales/zhCN/translation.json";
+import zhTW from "./locales/zhTW/translation.json";
+
+export const i18nconfig: InitOptions = {
+  fallbackLng: "en-CA",
+  debug: false,
+  resources: {
+    ar: { translation: ar }, // Arabic
+    bn: { translation: bn }, // Bengali
+    de: { translation: de }, // German
+    en: { translation: en }, // English
+    es: { translation: es }, // Spanish
+    fr: { translation: fr }, // French
+    he: { translation: he }, // Hebrew
+    hr: { translation: hr }, // Croatian
+    hu: { translation: hu }, // Hungarian
+    it: { translation: it }, // Italian
+    ja: { translation: ja }, // Japanese
+    pl: { translation: pl }, // Polish
+    pt: { translation: pt }, // Portugese
+    ro: { translation: ro }, // Romanian
+    sr: { translation: sr }, // Serbian
+    ru: { translation: ru }, // Russian
+    ta: { translation: ta }, // Tamil
+    tl: { translation: tl }, // Tagalog
+    tr: { translation: tr }, // Turkish
+    "zh-CN": { translation: zhCN }, // Chinese (Traditional)
+    "zh-TW": { translation: zhTW }, // Chinese (Simplified)
+  },
+  interpolation: {
+    escapeValue: false, // not needed for react as it escapes by default
+    format: function format(value, fmt, lng) {
+      if (value instanceof Date && fmt !== undefined) {
+        return value.toLocaleDateString(lng, {
+          year: "numeric",
+          month: "long",
+          day: "numeric",
+        });
+      }
+      return value;
+    },
+  },
+};
 
 i18n
-  // load translation using http -> see /public/locales (i.e. https://github.com/i18next/react-i18next/tree/master/example/react/public/locales)
-  // learn more: https://github.com/i18next/i18next-http-backend
-  .use(Backend)
   // detect user language
   // learn more: https://github.com/i18next/i18next-browser-languageDetector
   .use(LanguageDetector)
   // pass the i18n instance to react-i18next.
   .use(initReactI18next)
   // init i18next
   // for all options read: https://www.i18next.com/overview/configuration-options
-  .init({
-    fallbackLng: "en",
-    debug: false,
-
-    interpolation: {
-      escapeValue: false, // not needed for react as it escapes by default
-      format: function format(value, fmt, lng) {
-        if (value instanceof Date && fmt !== undefined) {
-          if (lng?.substring(0, 2) === "fr") {
-            return date_fnz_tz_format(value, fmt, { locale: frCA });
-          }
-          return date_fnz_tz_format(value, fmt, { locale: enCA });
-        }
-        return value;
-      },
-    },
+  .init(i18nconfig, function callback() {
+    // We use zh-CN for Simplified Chinese and zh-TW for Traditional Chinese
```

Comment:
Wow, i'm glad I've only ever had to support localizing English and French

Reply:
I only need this for Chinese because there are two versions Simplified and Traditional. We only have Latin script for Serbian if we get Cyrillic script we'll need to do this. Same with Punjabi which has two scripts which we don't have yet. Most other languages we don't need this. e.g. we don't implement separate translations for en-CA and en-US only difference is minor things like colour vs color.